cmd/kola/testiso.go

```diff
@@ -0,0 +1,314 @@
+					Enabled:  util.BoolToPtr(true),
+				},
+			},
+		},
+	}
+	var configstr string
+	if sdk.TargetIgnitionVersionFromName(metalimg) == "v2" {
+		ignc2, err := ignconverter.Translate3to2(config)
+		if err != nil {
+			return err
+		}
+		buf, err := json.Marshal(ignc2)
+		if err != nil {
+			return err
+		}
+		configstr = string(buf)
+	} else {
+		buf, err := json.Marshal(config)
+		if err != nil {
+			return err
+		}
+		configstr = string(buf)
+	}
+
+	tempdir, err := ioutil.TempDir("", "kola-testiso")
+	if err != nil {
+		return err
+	}
+
+	tftpdir := filepath.Join(tempdir, "tftp")
+	if err := os.Mkdir(tftpdir, 0777); err != nil {
+		return err
+	}
+
+	metalIsCompressed := !strings.HasSuffix(metalimg, ".raw")
+	metalname := metalimg
+	if !metalIsCompressed {
+		fmt.Println("Compressing metal image")
+		metalimgpath := filepath.Join(builddir, metalimg)
+		srcf, err := os.Open(metalimgpath)
+		if err != nil {
+			return err
+		}
+		defer srcf.Close()
+		metalname = metalname + ".gz"
+		destf, err := os.OpenFile(filepath.Join(tftpdir, metalname), os.O_RDWR|os.O_CREATE, 0755)
+		if err != nil {
+			return err
+		}
+		defer destf.Close()
+		cmd := exec.Command("gzip", "-1")
+		cmd.Stdin = srcf
+		cmd.Stdout = destf
+		if err := cmd.Run(); err != nil {
+			return errors.Wrapf(err, "running gzip")
+		}
+	}
+
+	mux := http.NewServeMux()
+	serveconfig := func(w http.ResponseWriter, r *http.Request) {
+		fmt.Println("Serving Ignition config")
+		io.WriteString(w, configstr)
+	}
+	mux.HandleFunc("/config.ign", serveconfig)
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return err
+	}
+	fserver := http.FileServer(http.Dir(tftpdir))
+	servecontent := func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(os.Stderr, "serving %s\n", r.URL)
+		fserver.ServeHTTP(w, r)
+	}
+	mux.HandleFunc("/", servecontent)
+	port := listener.Addr().(*net.TCPAddr).Port
+	go func() {
+		http.Serve(listener, mux)
+	}()
+	baseurl := fmt.Sprintf("http://%s:%d", tftpipaddr, port)
+
+	for _, name := range []string{kola.CosaBuild.BuildArtifacts.Kernel.Path, kola.CosaBuild.BuildArtifacts.Initramfs.Path} {
+		src, err := filepath.Abs(filepath.Join(builddir, name))
+		if err != nil {
+			return err
+		}
+		if err := os.Symlink(src, filepath.Join(tftpdir, name)); err != nil {
+			return err
+		}
+	}
+
+	kargs := []string{"rd.neednet=1", "ip=dhcp", "coreos.inst=yes", "coreos.inst.install_dev=vda"}
+	kargs = append(kargs, fmt.Sprintf("console=%s", consoleKernelArgument[kola.QEMUOptions.Board]))
+	kargs = append(kargs, fmt.Sprintf("coreos.inst.image_url=%s/%s", baseurl, metalname))
+	kargs = append(kargs, fmt.Sprintf("coreos.inst.ignition_url=%s/config.ign", baseurl))
+
+	kargsStr := strings.Join(kargs, " ")
+
+	var bootfile string
+	switch boottype {
+	case "pxe":
+		pxeconfigdir := filepath.Join(tftpdir, "pxelinux.cfg")
+		if err := os.Mkdir(pxeconfigdir, 0777); err != nil {
+			return err
+		}
+		pxeimages := []string{"pxelinux.0", "ldlinux.c32"}
```

Comment:
for s390, the pxelinux image has to be built from the kernel and initrd.
ref:https://github.com/coreos/coreos-assembler/pull/1139/commits/074e32836bdb108e158efdc2aa066ab8dd3fe8c9#diff-4cf8ac28258bdbad16564f473da21a71R104

Reply:
and the pxelinux parameters are passed a little differently which is annoying. 

@cgwalters since i will be testing s390x and ppc anyway i can make those tweaks if necessary

Reply:
OK yep I forgot to translate that part.  Let me hold a mutex on this for a bit and try to continue translation.

Reply:
OK mutex :lock: released - do you want to just post a `git diff` from any changes you need here?

(Do we have any docs on getting CoreOS maintainers access to ppc64le/s390x kube/OpenShift clusters set up to run cosa?)

Reply:
I went ahead and added you to [Co-authored-by](https://help.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-github).

Reply:
afaik there is just the one cluster for ppc64le/s390x which runs the rhcos jenkins pipeline. let me see if i can arrange something on this.

Reply:
thanks. tested on power and x390. looks good. on s390 needed a minor change:
https://gist.github.com/Prashanth684/5e49fd514471a58e21b739f48535d26e

Reply:
also we definitely can give you guys access to the ppc64le/s390x pipeline systems. Please get in touch with Andy McCrae ..he should be able to help out